common/app/common/templates/TemplateService.java

```diff
@@ -0,0 +1,28 @@
+package common.templates;
+
+import common.pages.PageData;
+import play.twirl.api.Html;
+
+/**
+ * Service that provides HTML pages, using some kind of template system.
+ */
+public interface TemplateService {
+
+    /**
+     * Injects the page data into the template with the given name.
+     * @param templateName name of the template
+     * @param pageData data to be injected in the template
+     * @return string of the HTML generated with the template and the given page data
+     */
+    String fill(final String templateName, final PageData pageData);
```

Comment:
The name fill sounds like you fill the the pageData, but it actually renders stuff. How about calling this "render"?